### PR TITLE
Helmholtz sound-hard scattering problem with absorbing boundary conditions

### PR DESCRIPTION
**Pull request by Ehaw04**
Hi, I added this tutorial

Purposes:

- Solve Helmholtz sound-hard scattering problem.

- Truncate unbounded domain via absorbing boundary condition

- Learn how to use Robin BC as absorbing boundary condition

### CODE REVIEW

File: examples/pinn_forward/Helmholtz_Sound_hard_ABC_2d.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
+"""Backend supported: tensorflow.compat.v1, tensorflow, pytorch"""
```
Comment (lululxvi) [inline examples/pinn_forward/Helmholtz_Sound_hard_ABC_2d.py:1]: Have you verified these backends?

Comment (Ehaw04) [reply examples/pinn_forward/Helmholtz_Sound_hard_ABC_2d.py:1]: Yes, the code was tested succesfully with the three backends.

File: docs/demos/pinn_forward/helmholtz.2d.sound.hard.abc.rst [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+
+.. math:: - u_{0,xx}-u_{0,yy} - k_0^2 u_0 = 0 \qquad \text{in}\qquad \Omega \cap D^{out}
+
+and
+
+.. math:: - u_{1,xx}-u_{1,yy} - k_0^2 u_1 = 0 \qquad\text{in}\qquad  \Omega \cap D^{out}
+
+The boundary conditions read:
+
+.. math::\gamma_1 u =  - \gamma_1 u^{inc} \qquad \text{on} \qquad\Gamma^{in}
```
Comment (lululxvi) [inline docs/demos/pinn_forward/helmholtz.2d.sound.hard.abc.rst:35]: miss an empty space after `.. math::`

File: docs/demos/pinn_forward/helmholtz.2d.sound.hard.abc.rst [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+
+  bcs = [bc0_inner, bc1_inner, bc0_outer, bc1_outer]
+
+
+Next, we define the weights for the loss function and generate the training and testing points.
+
+.. code-block:: python
+
+  loss_weights = [1, 1, weights, weights, weights, weights]
+  data = dde.data.PDE(geom, pde, bcs, num_domain= nx**2, num_boundary= 8 * nx, num_test= 5 * nx ** 2, solution = sol)
```
Comment (lululxvi) [inline docs/demos/pinn_forward/helmholtz.2d.sound.hard.abc.rst:188]: Line 188 doesn't have the correct format.

File: docs/demos/pinn_forward.rst [hunk 4]
```diff
@@ -30,6 +30,7 @@ Time-independent PDEs
    pinn_forward/eulerbeam
    pinn_forward/helmholtz.2d.dirichlet
    pinn_forward/helmholtz.2d.neumann.hole.rst
+   pinn_forward/helmholtz.2d.sound.hard.abc.rst
```
Comment (lululxvi) [inline docs/demos/pinn_forward.rst:33]: `pinn_forward/helmholtz.2d.sound.hard.abc`

File: docs/demos/pinn_forward/helmholtz.2d.sound.hard.abc.rst (hunk 2 above)
Comment (Ehaw04) [reply docs/demos/pinn_forward/helmholtz.2d.sound.hard.abc.rst]: Done!

File: docs/demos/pinn_forward/helmholtz.2d.sound.hard.abc.rst (hunk 3 above)
Comment (Ehaw04) [reply docs/demos/pinn_forward/helmholtz.2d.sound.hard.abc.rst]: Done!

File: docs/demos/pinn_forward.rst (hunk 4 above)
Comment (Ehaw04) [reply docs/demos/pinn_forward.rst]: Also done! thank you for your suggestions

File: docs/demos/pinn_forward/helmholtz.2d.sound.hard.abc.rst [hunk 5]
```diff
@@ -0,0 +1,226 @@
...
+
+  loss_weights = [1, 1, weights, weights, weights, weights]
+  data = dde.data.PDE(
+      geom,
+      pde,
+      bcs,
+      num_domain= nx**2,
+      num_boundary= 8 * nx,
+      num_test= 5 * nx ** 2,
+      solution = sol
```
Comment (lululxvi) [inline docs/demos/pinn_forward/helmholtz.2d.sound.hard.abc.rst:195]: Match the format of the python code
```python
      num_domain=nx**2,
      num_boundary=8 * nx,
      num_test=5 * nx**2,
      solution=sol,
```

Comment (Ehaw04) [reply docs/demos/pinn_forward/helmholtz.2d.sound.hard.abc.rst]: Done, thank you!

**Pull request merged**